avocado/core/parser.py

```diff
@@ -172,3 +178,94 @@ def finish(self):
             self.application.error(msg)
         # from this point on, config is a dictionary based on a argparse.Namespace
         self.config = vars(args)
+
+
+class HintParser:
+    def __init__(self, filename):
+        self.filename = filename
+        self.config = None
+        self.hints = []
+        self._parse()
+
+    def _get_args_from_section(self, section):
+        args = self.config.get(section, 'args')
+        if args == '$test':
+            return [args]
+        return args.split(',')
+
+    def _get_kwargs_from_section(self, section):
+        result = {}
+        kwargs = self.config.get(section, 'kwargs', fallback='')
+        for kwarg in kwargs.split(','):
+            if kwarg == '':
+                continue
+            key, value = kwarg.split('=')
+            result[key] = value
+        return result
+
+    def _get_resolutions_by_kind(self, kind, paths):
+        self.validate_kind_section(kind)
+
+        resolutions = []
+        uri = self._get_uri_from_section(kind)
+        args = self._get_args_from_section(kind)
+        kwargs = self._get_kwargs_from_section(kind)
+        success = ReferenceResolutionResult.SUCCESS
+        if uri == '$test':
```

Comment:
I understand the point about the variable being named `$test`, but, even though it's a word that makes sense to this specific case, it may not make sense to other cases. AFAICT, `$test` is really the path what was matched, so I'd "come clean" and use that as a variable name instead.

Reply:
@clebergnu do you mean `$testpath`?

Reply:
`$testpath` sounds good, I mean, better than both `$test` and `$path`.

Reply:
Fixed on v2